hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactFunction.java

```diff
@@ -114,6 +117,17 @@ private void doCompaction(String instantTime, CompactionOperation compactionOper
     collector.collect(new CompactionCommitEvent(instantTime, compactionOperation.getFileId(), writeStatuses, taskID));
   }
 
+  private void refresh() throws Exception {
+    HoodieTableMetaClient metaClient = writeClient.getHoodieTable().getMetaClient();
+    metaClient.reloadActiveTimeline();
+
+    // set table schema
+    CompactionUtil.setAvroSchema(conf, metaClient);
+
+    // refresh writeClient
+    writeClient = StreamerUtil.createWriteClient(conf, getRuntimeContext());
+  }
```

Comment:
No need to create the write client again, just update the schema of the write config, and in which case do we need the schema evolution, schema evolution is not supported now ~

Reply:
Ok, I will update this and waiting for schema evolution supported.

Reply:
I guess writeConfig.setSchema should work, so no need to rebuild it ~